base/int.jl

```diff
@@ -382,7 +382,7 @@ julia> leading_zeros(Int32(1))
 31
 ```
 """
-leading_zeros(x::BitInteger) = ctlz_int(x) % Int
+leading_zeros(x::BitInteger) = (ctlz_int(x) % Int)::Int
```

Comment:
Is this effective?
```julia
julia> code_warntype(leading_zeros, (Base.BitInteger,))
Variables
  #self#::Core.Compiler.Const(leading_zeros, false)
  x::Union{Int128, Int16, Int32, Int64, Int8, UInt128, UInt16, UInt32, UInt64, UInt8}

Body::Int64
1 ─ %1 = Base.ctlz_int(x)::Union{Int128, Int16, Int32, Int64, Int8, UInt128, UInt16, UInt32, UInt64, UInt8}
│   %2 = Base.Int(%1)::Int64
└──      return %2
```
Rather, I think that the `pad`s in `bin()` and so on need assertions.

Reply:
Ah, I see...
```julia
julia> code_warntype(leading_zeros, (Base.BitInteger,)) # 1.6.0-DEV.312
Variables
  #self#::Core.Compiler.Const(leading_zeros, false)
  x::Union{Int128, Int16, Int32, Int64, Int8, UInt128, UInt16, UInt32, UInt64, UInt8}

Body::Any
1 ─ %1 = Base.ctlz_int(x)::Any
│   %2 = (%1 % Base.Int)::Any
└──      return %2
```